driver/socketcall_to_syscall.c

```diff
@@ -155,8 +153,92 @@ int socketcall_code_to_syscall_code(int socketcall_code)
 		break;
 	}
 
-	/* if we are not able to convert the
-	 * socketcall code we return -1
+	/* If we cannot convert to a valid syscall id, there are 2 possibilities:
+	 * 1. the user provided the wrong socket call code.
+	 *	  In this case we will send a generic event at the end of this method
+	 * 2. The socket call code is defined but the corresponding
+	 *    syscall call is not defined. For example on s390x machines
+	 *    `SYS_ACCEPT` is defined but `__NR_accept` is not.
+	 * 	  In this way we will send the corresponding event.
+	 *
+	 * Known cases in which the socket call code is defined but
+	 * the corresponding syscall code is not:
+	 *
+	 * ----- s390x
+	 * - `SYS_ACCEPT` is defined but `__NR_accept` is not defined
+	 *
+	 * ----- x86 with CONFIG_IA32_EMULATION
+	 * - `SYS_ACCEPT` is defined but `__NR_accept` is not defined
+	 * - `SYS_SEND` is defined but `__NR_send` is not defined
+	 * - `SYS_RECV` is defined but `__NR_recv` is not defined
 	 */
-	return -1;
+	*is_syscall_return = false;
+
+	switch(socketcall_code)
+	{
+	case SYS_SOCKET:
+		return PPME_SOCKET_SOCKET_E;
+
+	case SYS_SOCKETPAIR:
+		return PPME_SOCKET_SOCKETPAIR_E;
+
+	case SYS_ACCEPT:
+		return PPME_SOCKET_ACCEPT_5_E;
+
+	case SYS_ACCEPT4:
+		return PPME_SOCKET_ACCEPT4_5_E;
```

Comment:
`ACCEPT4_5` or `ACCEPT4_6` or it makes no difference?

Reply:
Great catch!